feat(cloud-provisioning): Adds cloud_keystore_installation resource

rvelaVenafi opened the pull request:
Adds cloud_keystore_installation resource to provision a certificate to a cloud keystore (AKV, ACM, GCM)

Closes VC-32723

rvelaVenafi committed:
feat(cloud-provisioning): Adds cloud_keystore_installation resource

Adds cloud_keystore_installation resource to provision a certificate to a cloud keystore (AKV, ACM, GCM)

Closes VC-32723

luispresuelVenafi commented on go.mod:
```diff
@@ -3,9 +3,10 @@ module github.com/Venafi/terraform-provider-venafi
 
 require (
-	github.com/Venafi/vcert/v5 v5.6.5-0.20240516011016-3f129abb0388
+	github.com/Venafi/vcert/v5 v5.7.0-rc2.0.20240525040228-9a6c6b02a545
```
could you update here to use `v5.7.0`?

rvelaVenafi replied:
will do

rvelaVenafi replied:
unfortunately, 5.7.0 does not contain the structs required for terraform to work. I will wait until 5.7.1 is released to change it

sbanic-venafi commented on venafi/resource_cloudkeystore_installation.go:
```diff
@@ -0,0 +1,494 @@
+			},
+			cloudKeystoreInstallationCloudCertificateName: {
+				Type:        schema.TypeString,
+				Description: "ID of the parent Cloud Provider the keystore belongs to",
```
Is this description correct?

rvelaVenafi replied:
Thanks, changed the description to fit the attribute

rvelaVenafi committed:
feat(cloud-provisioning): Adds logging to new feature

Adds function to normalize certificate name for provisioning

Closes VC-32723

rvelaVenafi committed:
feat(debugging): Adds proper debugging options to provider

Adds makefile target to start provider with debug enabled

Closes VC-32723

rvelaVenafi committed:
feat(debugging): cloud_keystore_installation destroy

Adds conditional to invoke delete API for cloud_keystore_installation only when skip_retirement is true.

When skip_retirement is false, the related certificate will be retired from VCP and retirement will trigger machine identity deletion. In this case, the cloud_keystore_installation resource will only be deleted from terraform state

Closes VC-32723

rvelaVenafi committed:
feat: cloud_keystore_installation

Fixes an issue that swapped update and delete operation declaration in venafi_cloud_keystore_installation resource schema

Make key_password optional for service generated CSR. When not set, certificate, chain and private key will not be stored in state, and all private key related attributes will not be populated

Adds renew_required attribute to venafi_certificate resource to control when the resource should be destroyed and recreated as consequence of certificate remaining validity within expiration window

Makes key_password optional for venafi_certificate resource import

Updates expiration_window behavior. Now if certificate validity is within expiration window, the certificate will be renewed in the same terraform run as part of the plan. It is no longer needed to run terraform apply a second time for the certificate to be renewed

Updates some attribute descriptions to be consistent across the provider

Closes VC-32723
